core/src/main/java/org/elasticsearch/action/support/AutoCreateIndex.java

```diff
@@ -62,19 +63,21 @@ public boolean needToCheck() {
     }
 
     /**
-     * Should the index be auto created?
+     * Should the index be auto created? Returns {@code true} if it should be auto created, returns {@code false} if it already exists, and
+     * throws {@link IndexNotFoundException} if it can't be created.
      */
     public boolean shouldAutoCreate(String index, ClusterState state) {
```

Comment:
I'm not super sure about the name of this method with this behavior. The behavior seems fine for how the method is used, but the name is a bit funky now.